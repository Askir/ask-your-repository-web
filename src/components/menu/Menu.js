// @flow
import React from 'react';
import Popup from 'reactjs-popup';
import MenuIcon from './MenuIcon';
import MenuItems from './MenuItems';
import './Menu.scss';

const contentStyle = {
  background: "rgba(255,255,255,0",
  width: "80%",
  border: "none"
};

const overlayStyle = {
  background: 'rgba(255,255,255,0.98'
};

function Menu() {
  return (
    <Popup
      modal
      overlayStyle={overlayStyle}
      contentStyle={contentStyle}
      closeOnDocumentClick={false}
      trigger={open => <MenuIcon open={open} />}
    >
      {close => <MenuItems close={close} />}
    </Popup>
  );
}

export default Menu;
