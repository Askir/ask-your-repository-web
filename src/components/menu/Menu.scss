/* Menu Icon */

.burger-menu, .burger-menu.open {
  display: inline-block;
  cursor: pointer;
  position: fixed;
  top: 20px;
  left: 20px;
  z-index: 9999;
  background: var(--main-text-color);
  padding: 10px;
  border-radius: 25px;
  box-shadow: 0 2.5px 10px rgba(0, 0, 0, 1);
}

.burger-menu:hover {
  background: var(--medium-background-color);
}

.burger-menu .bar1, .bar2, .bar3 {
  width: 25px;
  height: 3px;
  background-color: #333;
  margin: 4px 0;
  transition: 0.4s;
}

.burger-menu.open .bar1 {
  -webkit-transform: rotate(-45deg) translate(-4px, 4px);
  transform: rotate(-45deg) translate(-4px, 4px);
}

.burger-menu.open .bar2 {
  opacity: 0;
}

.burger-menu.open .bar3 {
  -webkit-transform: rotate(45deg) translate(-6px, -6px);
  transform: rotate(45deg) translate(-6px, -6px);
}

/* Menu */

.menu {
  width: 100%;
  display: block;
  text-align: center;
  padding: 0px;
}
.menu ul {
  position: relative;
  top: 0px;
  font-size: 30px;
  padding: 0px;
  list-style-type: none;
  line-height: 2.5em;
}
.menu li a{
  margin: 0px 0px;
  padding: 0;
  cursor: pointer;
  border: none;
  color: var(--dark-text-color);
}
.menu li a:hover {
  color: var(--accent-color-1);
}
