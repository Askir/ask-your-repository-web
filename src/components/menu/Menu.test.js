// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Menu from './Menu';

describe('<Menu />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Menu />).dive();
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
