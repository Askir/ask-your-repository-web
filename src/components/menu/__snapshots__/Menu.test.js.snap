// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders correctly 1`] = `
Array [
  <Ref
    innerRef={[Function]}
    key="R"
  >
    <MenuIcon
      key="T"
      onClick={[Function]}
      open={false}
    />
  </Ref>,
  "",
  "",
  "",
]
`;
